### PR TITLE
Refactor CVE fix skill into shell scripts

### PR DESCRIPTION
**Pull request by dfarrell07**
Extract all inline bash from skills/cve-fix/SKILL.md into shellcheck-linted scripts. The SKILL.md becomes a thin wrapper that delegates to fix-all.sh.

The deterministic orchestrator (fix-all.sh) handles the full flow: detect config, create fix branch, scan with grype, fix each CVE, verify, commit, run tests. After the deterministic phase, it invokes a Claude subagent (review.sh) to review all outcomes and handle edge cases like breaking changes or persistent CVEs.

Add make cve-fix target for standalone use outside the skill.

See SEP-0032 for design details.

## Summary by CodeRabbit

* **New Features**
  * End-to-end CVE remediation tooling: detect, scan, apply fixes, generate commits/PR hints, review assistance, ignore/locate utilities, and cleanup commands.

* **Documentation**
  * Expanded usage guides, reviewer prompt template, automated-run examples, troubleshooting/network guidance, and abbreviated commit-message mappings for CVE fixes.

* **Tests**
  * CI step and unit/integration tests added to validate the CVE workflow and scripts.

* **Chores**
  * Make targets and skill wrapper updated to run/clean the CVE workflow.

[Review Change Stack](https://app.coderabbit.ai/change-stack/submariner-io/shipyard/pull/2383?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,6 +65,10 @@ deploy deploy-latest e2e upgrade-e2e: package/.image.nettest
 
 include Makefile.dapper
 
+# CVE fix scripts
+cve-fix:
+	./scripts/cve/fix-all.sh $(or $(REPO),.) $(or $(BRANCH),$(shell git branch --show-current))
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply Makefile]: Fixed. Added double quotes around both $(or) expansions.

File: scripts/cve/detect.sh [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+GENERATED_FILE="$GENERATED_FILE"
+DIFF_IGNORE_ARGS="$DIFF_IGNORE_ARGS"
+NEEDS_BUILD_FOR_SCAN=$NEEDS_BUILD_FOR_SCAN
+CONTAINER_CMD="$CONTAINER_CMD"
+HAS_LOCAL_GRYPE=$HAS_LOCAL_GRYPE
+SHIPYARD_IMAGE="$SHIPYARD_IMAGE"
+SHIPYARD_GO_VERSION="$SHIPYARD_GO_VERSION"
+FETCH_FAILED=$FETCH_FAILED
+CVE_SCRIPTS="$SCRIPT_DIR"
+EOF
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/detect.sh:158]: Won't fix. All values are controlled (paths from pwd, git branch names, hardcoded strings like docker/true/false). No user-supplied input reaches the state file.

File: scripts/cve/fix-all.sh [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+# shellcheck source=lib.sh
+source "$SCRIPT_DIR/lib.sh"
+
+# Clean up on exit: revert uncommitted changes and remove state file
+# shellcheck disable=SC2317,SC2329 # invoked via trap EXIT
+cleanup() {
+  git checkout -- . 2>/dev/null || true
+  rm -f "${STATE_FILE:-}"
+}
+trap cleanup EXIT
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Fixed differently. Added a precondition check in detect.sh that refuses to run on a dirty working tree (git diff --quiet && git diff --cached --quiet), preventing the cleanup trap from ever discarding uncommitted work.

File: scripts/cve/fix-all.sh [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      severity: .vulnerability.severity,
+      type: (if .artifact.name == "stdlib" then "stdlib" else "package" end)
+    }
+  ] |
+  group_by(.pkg) |
+  map({
+    pkg: .[0].pkg,
+    type: .[0].type,
+    fixedIn: (map(.fixedIn) | sort | last),
+    cves: map(.cve),
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Fixed. Changed sort | last to sort_by(split(".") | map(tonumber)) | last.

File: scripts/cve/fix-all.sh [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+make unit || echo "WARNING: make unit failed"
+
+echo ""
+echo "Final scan..."
+"$SCRIPT_DIR/scan.sh" "$STATE_FILE" 2>/dev/null || true
+
+# --- Phase 2: Agent Review ---
+
+banner "Phase 2: Agent Review"
+"$SCRIPT_DIR/review.sh" "$STATE_FILE" "$FIX_SUMMARY" || true
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Won't fix. By design, the exit code reflects the deterministic phase. The agent review may fix additional CVEs, but parsing its output would be fragile. Exit 2 means "human should check," which is correct when there were unresolved CVEs.

File: scripts/cve/fix-package.sh [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  fi
+done < <(find_gomods)
+
+# Snapshot go directives before update (for breaking-change detection)
+GO_BEFORE=""
+while IFS= read -r GOMOD; do
+  GO_VER=$(grep '^go ' "$GOMOD" 2>/dev/null | awk '{print $2}')
+  [[ -n "$GO_VER" ]] && GO_BEFORE+="$GOMOD:$GO_VER "
+done < <(find_gomods)
+K8S_BEFORE=$(grep 'k8s.io/' go.mod 2>/dev/null | head -1 | grep -oP 'v0\.\K[0-9]+' || echo "")
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh]: Fixed. Changed from grepping root go.mod only to grep -h k8s.io/client-go across all go.mod files via find_gomods. Uses sort -un to compare unique minor version sets.

File: scripts/cve/fix-package.sh [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+# Update in all go.mod files that contain this package
+while IFS= read -r GOMOD; do
+  MODDIR=$(dirname "$GOMOD")
+  if grep -q "$PACKAGE" "$GOMOD" 2>/dev/null; then
+    if [[ "$MODDIR" == "." ]]; then
+      go get "${PACKAGE}@v${VERSION}" && go mod tidy
+    else
+      go -C "$MODDIR" get "${PACKAGE}@v${VERSION}" && go -C "$MODDIR" mod tidy
+    fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh]: Won't fix. grep on go.mod is reliable, fast, and doesn't require Go toolchain context per module. go list -m -json is slower and adds no practical accuracy for our use case.

File: scripts/cve/fix-package.sh [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+done < <(find_gomods)
+
+if [[ "$STILL_VULNERABLE" == "true" ]]; then
+  echo "NEEDS_REVIEW: $PACKAGE — CVE persists after update to v$VERSION"
+  git checkout -- . || echo "ERROR: Could not rollback changes" >&2
+  exit 3
+fi
+
+# Stage all changed go.mod/go.sum files (some repos have extra modules like coredns/)
+git diff --name-only | grep -E 'go\.(mod|sum)$' | xargs -r git add
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh]: Fixed. Added || true to the pipeline to prevent ERR trap on no matches under set -euo pipefail.

File: scripts/cve/fix-stdlib.sh [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    OLDEST=$(printf '%s\n' "$SHIPYARD_GO" "$GO_VERSION" | sort -V | head -1)
+    if [[ "$OLDEST" == "$SHIPYARD_GO" ]] && [[ "$SHIPYARD_GO" != "$GO_VERSION" ]]; then
+      echo "NOTE: Shipyard build image has Go $SHIPYARD_GO but go.mod now requires $GO_VERSION"
+      echo "CI may fail until Shipyard is updated with a newer Go version."
+    fi
+  fi
+fi
+
+# Stage all changed go.mod/go.sum files
+git diff --name-only | grep -E 'go\.(mod|sum)$' | xargs -r git add
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-stdlib.sh]: Won't fix. All Submariner repos already have go.sum in every module. go mod tidy modifies existing files, doesn't create new ones. The || true pipeline fix was applied for the same pipefail safety issue.

File: scripts/cve/lib.sh [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+  if [[ -n "$CONTAINER_CMD" ]]; then
+    if [[ "$fresh" == "true" ]]; then
+      $CONTAINER_CMD volume create grype-db >/dev/null 2>&1 || true
+      $CONTAINER_CMD run --pull=always --rm \
+        -v grype-db:/root/.cache/grype anchore/grype:latest db update >&2 || true
+    fi
+    if $CONTAINER_CMD run --rm \
+      -v grype-db:/root/.cache/grype \
+      -v "$(pwd)":/src \
+      anchore/grype:latest /src --config /src/.grype.yaml -o "$output_format"; then
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/lib.sh:79]: Won't fix. --pull=always already pulls the latest grype image on --fresh. The DB update runs separately. The scanner is refreshed.

File: scripts/cve/lib.sh [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    exclude_line=$(grep -n '^exclude:' "$file" | head -1 | cut -d: -f1)
+    {
+      head -n "$((exclude_line - 1))" "$file"
+      printf '%s\n' "$entry"
+      tail -n +"$exclude_line" "$file"
+    } > "${file}.tmp"
+    mv "${file}.tmp" "$file"
+  else
+    printf '\n%s\n' "$entry" >> "$file"
+  fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/lib.sh:154]: Won't fix. insert_grype_ignore is called once per CVE by ignore.sh. The orchestrator tracks what's been fixed/ignored, so duplicate calls don't happen in practice.

File: scripts/cve/locate.sh [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+STATE_FILE="${1:?Usage: locate.sh STATE_FILE PACKAGE}"
+PACKAGE="${2:?Usage: locate.sh STATE_FILE PACKAGE}"
+load_state "$STATE_FILE"
+
+echo "=== Locating: $PACKAGE ==="
+
+# Find in all go.mod files
+FOUND_IN=""
+while IFS= read -r GOMOD; do
+  if grep -q "$PACKAGE" "$GOMOD" 2>/dev/null; then
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/locate.sh]: Won't fix. Go module paths (github.com/docker/docker, k8s.io/client-go) don't contain regex metacharacters. grep -q is safe and readable.

File: scripts/cve/review.sh [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+# Current scan state
+echo "Running scan for review evidence..."
+# shellcheck disable=SC2119 # no-args call is intentional (default table format)
+CURRENT_SCAN=$(run_grype 2>&1) || CURRENT_SCAN="(scan failed)"
+
+# Locate info for any NEEDS_REVIEW packages
+LOCATE_OUTPUT=""
+NEEDS_REVIEW_PKGS=$(echo "$FIX_SUMMARY" | grep "^NEEDS_REVIEW:" | sed 's/NEEDS_REVIEW: \([^ ]*\).*/\1/' || true)
+for PKG in $NEEDS_REVIEW_PKGS; do
+  LOCATE_OUTPUT+="$(bash "$SCRIPT_DIR/locate.sh" "$STATE_FILE" "$PKG" 2>&1 || true)"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/review.sh:40]: Won't fix. The review agent receives all items in FIX_SUMMARY and handles them uniformly. It understands both stdlib and package types from context.

File: scripts/cve/review.sh [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+  echo "$FIX_SUMMARY"
+  exit 0
+fi
+
+OUTPUT=$(claude -p "$PROMPT" \
+  --print \
+  --model sonnet \
+  --allowedTools "Bash" \
+  --dangerously-skip-permissions \
+  2>&1) || true
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/review.sh:67]: Won't fix. The review agent has Bash access and can run git log/diff if needed. Pre-fetching commit diffs would add more data to an already information-dense prompt.

File: scripts/cve/fix-all.sh [hunk 15]
```diff
@@ -0,0 +1,204 @@
...
+        ;;
+    esac
+  fi
+  rm -f "$FIX_LOG"
+done <<< "$CVE_LINES"
+
+# Verify
+echo ""
+echo "Running unit tests..."
+make unit || echo "WARNING: make unit failed"
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Good catch. Fixed. make unit failures now increment REVIEW_COUNT and add to FIX_SUMMARY, so the exit code correctly reflects test failures.

File: scripts/cve/fix-package.sh [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+done < <(find_gomods)
+
+# Snapshot go directives before update (for breaking-change detection)
+GO_BEFORE=""
+while IFS= read -r GOMOD; do
+  GO_VER=$(grep '^go ' "$GOMOD" 2>/dev/null | awk '{print $2}')
+  [[ -n "$GO_VER" ]] && GO_BEFORE+="$GOMOD:$GO_VER "
+done < <(find_gomods)
+# shellcheck disable=SC2046 # word splitting is intentional (multiple file args)
+K8S_BEFORE=$(grep -h 'k8s.io/client-go' $(find_gomods) 2>/dev/null | grep -oP 'v0\.\K[0-9]+' | sort -un | tr '\n' ' ')
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh:40]: Won't fix. Our CI runs on Linux with GNU grep. The -oP flag is standard there. macOS portability isn't a requirement for this project.

File: scripts/cve/locate.sh [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+  echo "ERROR: $PACKAGE not found in any go.mod" >&2
+  exit 1
+fi
+echo "Found in: $FOUND_IN"
+
+# Check for replace directives across all go.mod files
+PKG_BASENAME=$(basename "$PACKAGE")
+REPLACE_HITS=""
+while IFS= read -r GOMOD; do
+  HITS=$(grep "replace.*${PKG_BASENAME}" "$GOMOD" 2>/dev/null || true)
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/locate.sh]: Good catch. Fixed in both fix-package.sh and locate.sh. Now uses the full package path instead of basename for replace directive search.

File: scripts/cve/fix-package.sh [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+STATE_FILE="${1:?Usage: fix-package.sh STATE_FILE PACKAGE VERSION CVE_ID [CVE_ID...]}"
+PACKAGE="${2:?Missing PACKAGE}"
+VERSION="${3:?Missing VERSION}"
+shift 3
+CVE_IDS=("$@")
+[[ ${#CVE_IDS[@]} -gt 0 ]] || { echo "ERROR: At least one CVE_ID required" >&2; exit 1; }
+
+load_state "$STATE_FILE"
+
+trap 'git checkout -- . 2>/dev/null || true' ERR
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh]: Good catch. Fixed in both fix-package.sh and fix-stdlib.sh. ERR trap now runs git reset HEAD before git checkout to clean up staged files.

File: scripts/cve/fix-package.sh [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+while IFS= read -r GOMOD; do
+  GO_AFTER=$(grep '^go ' "$GOMOD" 2>/dev/null | awk '{print $2}')
+  [[ -z "$GO_AFTER" ]] && continue
+  # Find the before version for this go.mod
+  for PAIR in $GO_BEFORE; do
+    if [[ "${PAIR%%:*}" == "$GOMOD" ]]; then
+      GO_WAS="${PAIR#*:}"
+      if [[ "$(echo "$GO_WAS" | cut -d. -f1-2)" != "$(echo "$GO_AFTER" | cut -d. -f1-2)" ]]; then
+        BREAKING="$GOMOD: Go $GO_WAS -> $GO_AFTER"
+      fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/fix-package.sh:64]: Fixed. Changed to ${BREAKING:+$BREAKING; } accumulation pattern, matching the K8S check.

File: scripts/cve/fix-all.sh [hunk 20]
```diff
@@ -0,0 +1,207 @@
...
+
+echo ""
+echo "Cleaning build artifacts..."
+make clean >/dev/null 2>&1 || true
+
+echo "Scanning for CVEs..."
+SCAN_LOG=$(mktemp)
+"$SCRIPT_DIR/scan.sh" "$STATE_FILE" --fresh --json > "$SCAN_LOG" || true
+SCAN_JSON=$(cat "$SCAN_LOG")
+rm -f "$SCAN_LOG"
```
Comment (skitt) [inline scripts/cve/fix-all.sh]:
```suggestion
SCAN_JSON=$("$SCRIPT_DIR/scan.sh" "$STATE_FILE" --fresh --json || true)
```

File: scripts/cve/fix-all.sh [hunk 21]
```diff
@@ -0,0 +1,207 @@
...
+
+MATCH_COUNT=$(printf '%s\n' "$SCAN_JSON" | jq '.matches | length' 2>/dev/null) || MATCH_COUNT=""
+if [[ -z "$MATCH_COUNT" ]]; then
+  echo "ERROR: Scan failed or produced invalid output." >&2
+  echo "Check scanner availability (grype or docker/podman)." >&2
+  exit 1
+fi
```
Comment (skitt) [inline scripts/cve/fix-all.sh]:
```suggestion
if ! MATCH_COUNT=$(printf '%s\n' "$SCAN_JSON" | jq '.matches | length' 2>/dev/null); then
  echo "ERROR: Scan failed or produced invalid output." >&2
  echo "Check scanner availability (grype or docker/podman)." >&2
  exit 1
fi
```

**Review by skitt — CHANGES_REQUESTED** [review]
I’ve started reviewing this but it’s going to take me a while, so I’ll leave comments as I go and let you know once it’s all done.

File: scripts/cve/fix-all.sh (hunk 20 above)
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Done, thanks.

File: scripts/cve/fix-all.sh (hunk 21 above)
Comment (dfarrell07) [reply scripts/cve/fix-all.sh]: Adopted the `if !` pattern, thanks. Added `|| [[ -z "$MATCH_COUNT" ]]` because jq exits 0 on empty input with no output, so `if !` alone misses the case where scan.sh produces nothing (e.g., no scanner available).

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai What of your feedback is unaddressed? Why haven't you approved?

**Comment by dfarrell07** [comment]
@coderabbitai Both remaining items addressed:

1. **`docker kill $(docker ps -q)` kills all containers** — Fixed. Added `scripts/cve/clean.sh` with scoped cleanup that only targets `grype:latest` and `fix-*-cves-*` containers. Referenced from SKILL.md and `make cve-clean`.

2. **`FOUND_IN` as space-joined string** — Won't fix. `find_gomods` returns paths like `go.mod`, `tools/go.mod` — Go module paths can't contain spaces.

File: scripts/cve/fix-package.sh [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+# Update in all go.mod files that contain this package
+while IFS= read -r GOMOD; do
+  MODDIR=$(dirname "$GOMOD")
+  if grep -q "$PACKAGE" "$GOMOD" 2>/dev/null; then
+    echo "Updating $PACKAGE in $GOMOD..."
+    if [[ "$MODDIR" == "." ]]; then
+      go get "${PACKAGE}@v${VERSION}" && go mod tidy
+    else
+      go -C "$MODDIR" get "${PACKAGE}@v${VERSION}" && go -C "$MODDIR" mod tidy
+    fi
```
Comment (skitt) [inline scripts/cve/fix-package.sh]: `go -C .` works fine:


```suggestion
    go -C "$MODDIR" get "${PACKAGE}@v${VERSION}" && go -C "$MODDIR" mod tidy
```

File: scripts/cve/fix-stdlib.sh [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+  echo "NEEDS_REVIEW: stdlib — would upgrade Go $OLD_GO -> $GO_VERSION (minor version change)"
+  exit 2
+fi
+
+# Check host Go version is sufficient
+HOST_GO=$(go version | grep -oP 'go\K[0-9]+\.[0-9]+\.[0-9]+')
+if [[ -n "$HOST_GO" ]] && [[ "$(printf '%s\n' "$HOST_GO" "$GO_VERSION" | sort -V | head -1)" == "$HOST_GO" ]] && [[ "$HOST_GO" != "$GO_VERSION" ]]; then
+  echo "NEEDS_REVIEW: stdlib — host Go $HOST_GO is older than required $GO_VERSION"
+  echo "Update Go: go install golang.org/dl/go${GO_VERSION}@latest && go${GO_VERSION} download"
+  echo "Or set GOTOOLCHAIN=auto to allow automatic toolchain downloads."
```
Comment (skitt) [inline scripts/cve/fix-stdlib.sh:40]: This advice isn’t usable in this context, is it? The check above will always fail even if `GOTOOLCHAIN` allowed downloading a newer version. Perhaps the `if` could check for `GOTOOLCHAIN=auto` as well.

File: scripts/cve/fix-stdlib.sh [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+echo "Updating go directive and tidying modules..."
+while IFS= read -r GOMOD; do
+  MODDIR=$(dirname "$GOMOD")
+  if [[ "$MODDIR" == "." ]]; then
+    go mod edit -go="${GO_VERSION}"
+    go mod tidy
+  else
+    go -C "$MODDIR" mod edit -go="${GO_VERSION}"
+    go -C "$MODDIR" mod tidy
+  fi
```
Comment (skitt) [inline scripts/cve/fix-stdlib.sh]:
```suggestion
  go -C "$MODDIR" mod edit -go="${GO_VERSION}"
  go -C "$MODDIR" mod tidy
```

File: scripts/cve/lib.sh [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+# Find all go.mod files in the repo (excluding vendor and gitignored dirs)
+find_gomods() {
+  git ls-files --cached --others --exclude-standard '*/go.mod' 'go.mod' 2>/dev/null || \
+    find . -name go.mod -not -path '*/vendor/*'
+}
+
+# Remove artifacts added by go mod tidy from all go.mod files:
+# - toolchain directive (we pin Go version via Shipyard image, not toolchain)
```
Comment (skitt) [inline scripts/cve/lib.sh]: Let’s ignore this for this PR, but this is inconsistent with the approach used in `fix-stdlib.sh`.

File: scripts/cve/locate.sh (hunk 12 above)
[the remark this answers is not in the document]
Comment (skitt) [reply scripts/cve/locate.sh]: `grep -qF` seems just as readable to me, and Go module paths can include `.`.

**Review by skitt — APPROVED** [review]
Nothing major, this can go in as-is.

File: scripts/cve/fix-package.sh (hunk 22 above)
Comment (dfarrell07) [reply scripts/cve/fix-package.sh]: Fixed — removed the if/else in fix-package.sh, fix-stdlib.sh, and locate.sh.

File: scripts/cve/fix-stdlib.sh (hunk 23 above)
Comment (dfarrell07) [reply scripts/cve/fix-stdlib.sh:40]: Added `[[ "${GOTOOLCHAIN:-}" != *auto* ]]` to the condition so it's skipped when GOTOOLCHAIN would handle the download. Covers both `auto` and `go1.X.Y+auto`.

File: scripts/cve/fix-stdlib.sh (hunk 24 above)
Comment (dfarrell07) [reply scripts/cve/fix-stdlib.sh]: Fixed — same change as fix-package.sh.

File: scripts/cve/lib.sh (hunk 25 above)
Comment (dfarrell07) [reply scripts/cve/lib.sh]: Not 100% sure what you mean here — if it's that the comment was misleading (fix-stdlib.sh does manage the go directive via `go mod edit`, so "we pin Go version via Shipyard image" overstated things), updated it to say the Shipyard image controls the build toolchain specifically. If it's about sed vs `go mod edit` being inconsistent tools for modifying go.mod, added a note that `go mod edit` has no `-droptoolchain` flag so sed is the only option. Happy to revisit if you meant something else.

File: scripts/cve/locate.sh (hunk 12 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/locate.sh]: Fixed — applied `-F` here and to all other bare-package grep calls in locate.sh, fix-package.sh, and fix-all.sh. Left `replace.*${PACKAGE}` patterns as regex.

**Comment by dfarrell07** [comment]
@coderabbitai review

File: scripts/cve/review.sh (hunk 13 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/review.sh:40]: Fixed. Non-package tokens like "stdlib" and "unit" are now skipped — they don't contain dots, so they're filtered before reaching locate.sh.

File: scripts/cve/review.sh (hunk 14 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/review.sh:67]: Won't fix, but improved. The review agent intentionally has write access — it runs fix-package.sh, ignore.sh, etc. to handle CVEs the deterministic phase couldn't fix. Removed --dangerously-skip-permissions since --allowedTools "Bash" is sufficient.

File: scripts/cve/lib.sh (hunk 10 above)
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/lib.sh:79]: Previous reply was incorrect — code used --pull=missing, not --pull=always. Fixed to --pull=always so --fresh genuinely refreshes the scanner image.

File: .agents/workflows/cve-fix.md [hunk 26]
```diff
@@ -1,5 +1,16 @@
 #### Fixing CVEs
 
+**Automated:** Use `/cve-fix` in Claude Code or `make cve-fix` from shipyard:
+
+```bash
+/cve-fix release-0.23 ../submariner-operator
+make cve-fix REPO=../submariner BRANCH=release-0.23
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply .agents/workflows/cve-fix.md:7]: Intentional — they're independent examples showing two invocation methods, each targeting a different repo to demonstrate the tools work with any Submariner repo.

**Comment by dfarrell07** [comment]
@coderabbitai do you have any more pending feedback? review again

File: scripts/cve/lib.sh [hunk 27]
```diff
@@ -0,0 +1,161 @@
...
+    local -a env_args=()
+    if [[ "$no_update" == "true" ]]; then
+      env_args=(-e GRYPE_DB_AUTO_UPDATE=false -e GRYPE_DB_VALIDATE_AGE=false)
+    fi
+    if $CONTAINER_CMD run --rm "${env_args[@]}" \
+      -v grype-db:/.cache/grype \
+      -v "$(pwd)":/src \
+      anchore/grype:latest /src --config /src/.grype.yaml -o "$output_format"; then
+      return 0
+    fi
```
[the remark this answers is not in the document]
Comment (dfarrell07) [reply scripts/cve/lib.sh:81]: The DB update step (line 67) already pulls the latest anchore/grype:latest with --pull=always. The scan on line 76 uses the same image tag, which is now cached from that pull. Adding --pull=always to the scan would just pull the same image a second time.

**Comment by dfarrell07** [comment]
@coderabbitai review

**Comment by dfarrell07** [comment]
@coderabbitai why haven't you voted approve
